src/js/data.js

```diff
@@ -270,6 +269,17 @@ DataFactory.prototype = {
   }
 }
 
+var DataError = function(message) {
```

Comment:
I need more persuasion. This error is specific to this code module/library. Other modules which have other custom errors might have similar. The library they relate to is more important than the function they serve. Locally defined Error/Exception subclasses is a pretty common pattern in library development